### ISSUE TITLE
[FR] OSARA's FX parameters: Switch from one parameter to another with CTRL+Tab and CTRL+Shift+Tab

### ISSUE THREAD

**Issue by Aleskis** [jcsteh/osara#691]
Hi,

Would it be possible to speed up the passage of contiguous parameters without having to go back through the parameter list thank to CTRL+Tab and CTRL+Shift+Tab ?
If we take the example of ReaSynth, we could imagine that while being in the Attack parameter values, CTRL+Tab jumps to the Release parameter values. This would dispense with the need to shift+tab, down arrow then tab to access the rellease values.
I mention the CTRL+Tab combination, but it could be left and right arrows that have the same effect as up and down, or something else that would lead to the same behavior.

Thanks for the great work on OSARA

**Comment by jcsteh** [comment]
Because the parameters aren't tabs, control+tab/control+shift+tab feels
pretty weird here. For the same reason, you wouldn't be told which
parameter you had switched to. I could label the slider with the name of
the parameter to deal with that, but that would also mean that tabbing from
the combo box to the slider would report the parameter name, which
increases verbosity slightly in that case. All in all, I think this would
be pretty clunky, but I'm keen to hear @ScottChesworth's thoughts.

**Comment by ScottChesworth** [comment]
I like the idea a lot, but agree that Control+Tab feels odd. Left and Right arrows could be a goer though?

**Comment by jcsteh** [comment]
Interestingly, I always use left/right to adjust that slider, not up/down. Assuming I'm not the only one, that could be a very difficult change for people to get used to.

Even aside from that, though, thoughts on the verbosity part?

**Comment by jennykbrennan** [comment]
I also really like this idea. It would speed up work in the parameter list considerably. The constant back and forth from the list and changing its value is a big part of why I don't use it as much as I might like.
For me, right and left arrow would feel very natural. But I do see that differentiating the up/down from the left/right might be strange when that's what is the normal. I use up/down as a rule.
With that said, the verbosity might be a bigger issue. I might be talking out of my posterior on this but could that be an Osara configuration toggle?
The improvement in speed would really only happen if there's no extra speach. But if it's a choice, I'm all for it. 
In plugins i use often and where the parameter list is familiar such as the reaEQ or Reacomp, the there would be little need to hear the parameter name.

**Comment by pitermach** [comment]
+1 from me, whether it'll be CTRL+Tab, right/left or some other key combination. THis would especially improve efficiency on Mac where the popup button requires more keystrokes to change parameters like I mentioned in #534

Regarding verbosity, I'm not sure what the various screen readers would speak if the label and value change under the cursor, so how about when you press a key OSARA would just speak the name and value of the parameter that was just switched to?

**Comment by ScottChesworth** [comment]
Are the sliders vertical or horizontal in the GUI, @jcsteh?

**Comment by mattymc2** [comment]
Could be something like control plus one through 0 to select one of the 10 possible plug-ins instantiated. I know you can have more than that but if you have 10 plug-ins on your track he probably need to rethink what the hell you were doing LOL. I imagine this will probably interfere with the main windows keystrokes though. I have no idea how this stuff works. I know for me I always remember what my effects Chain is  so it would kind of be more intuitive to be able to jump straight to the plug-in I want to edit

**Comment by ScottChesworth** [comment]
@mattymc2, that was a whole post without any sign of a burn. Are you ok hun? :)
For now this is about making navigation through params of a single plug-in more efficient. What you've proposed would be useful though. Can you open a separate issue?

**Comment by mattymc2** [comment]
Aaa crap sorry.  That’s what I get for not reading everything properly and filling in gaps for myself. Sorry guys I’ll raise another issue

**Comment by Aleskis** [comment]
Hey all,

Some thoughts:
On the equality of functionality between Windows and Mac for OSARA, ideally, it should be 100% identical. That said, given the knowledge and testing constraints of the code, there are already differences between the two platforms.
In fact, the argument that we don't know how to do it on the Mac doesn't seem to me to be so valid to refuse to do it on Windows.
On the question of verbosity when changing a parameter, I am also in favor of stating the name of the parameter to avoid bad manipulations. In my opinion, this extra verbosity is not a problem, since it can be interrupted with an arrow stroke to move the parameter one step. It would be annoying if it were not possible to interrupt it to get the next data, namely the value of the parameter.

In short, there are certainly constraints, but the benefits seem to me to outweigh the costs.
Cheers,

**Comment by ScottChesworth** [comment]
> In fact, the argument that we don't know how to do it on the Mac doesn't seem to me to be so valid to refuse to do it on Windows.
> 
I'm pretty sure every Mac user would disagree. We're cross platform by design and try to keep differences to a minimum, in line with Reaper itself.

**Comment by Aleskis** [comment]
I'm not saying that mac users should be left on the sidelines, I'm just saying that a feature idea has never been sacrificed because we didn't know how to do it for one of the two platforms.
As a Mac user myself, I'm still very attached to the idea of a universally compatible system.
I often see software editors who, by integrating a new feature for one OS first, leave themselves some extra time to bring it to the other system a little later.
I'm sorry if I gave the impression that the Mac was of lesser importance.

**Comment by jennykbrennan** [comment]
Now that this possibility is being discussed, I'm really wanting it to happen. What keys might be used really doesn't matter. As Control tab and Control shift tab is available on the Mac, it does seem like a good choice for that reason alone. It's the functionality I'm quite excited about. We all have our preferences and i think making everyone happy is simply just a distant dream. But it's not going to ruin my world to learn a keystroke that feels a bit awkward to start with. :) 
And I do have to agree that verbosity is probably not even a problem. I'm one who keeps saying to everyone, to everyone's annoyance, that knowledge is king. I should live  as I preech I suppose. :D. And as mentioned, a new selection interrupts speech, so there is no issue in the  end.
I just don't want the main idea to be sidetracked. especially by myself. 
So whatever reservations I spouted, quite thoughtlessly, can summarily be ignored. 
Cheers. .  :)

**Comment by ScottChesworth** [comment]
> a new selection interrupts speech, so there is no issue in the end.
> 
Are we certain that's the case on Mac? I know our reports don't always interrupt, hence the longstanding title bar bottleneck. Will a bunch of reports in a row interrupt each other if they're all from OSARA, "jcsteh?

I really like the ideas being floated here, but IMO we should be treading carefully around anything that creates disparity from platform to platform and screen reader to screen reader, what with the params dialog being such a widely used feature.
